src/components/Experience.ts

```diff
@@ -25,15 +40,103 @@ export class Experience<TGameStartr extends FullScreenPokemon> extends Component
     }
 
     /**
-     * Gives experience to a specified pokemon
+     * Gives experience to a specified pokemon.
      *
      * @param pokemon   The pokemon to give experience to.
      * @param experience   The amount of experience to give.
      */
     public gainExperience(pokemon: IPokemon, experience: number): void {
         pokemon.experience += experience;
-        if (pokemon.experience >= this.gameStarter.equations.experienceStarting(pokemon.title, pokemon.level + 1)) {
+        if (pokemon.experience >= this.gameStarter.equations.experienceStarting(pokemon.title, pokemon.level + 1)
+            && pokemon.statistics.health.current !== 0) {
```

Comment:
This way means if someone calls `gainExperience` on a fainted Pokemon with enough points to get it to the next level, the Pokemon's data is suddenly in a corrupted state.